app/controllers/application_controller.rb

```diff
@@ -30,6 +30,20 @@ def upgrade_warning
     basecamp_auth_check
   end
 
+  def filtered_agent_return_link(options = {})
+    case ret = params[:return].presence || options[:return]
+      when "show"
```

Comment:
This is what I usually use, although I don't feel strongly.  I find it easier to read if the `when` statements are indented one level from the `case`.